app/javascript/packages/i18n/index.ts

```diff
@@ -35,6 +35,25 @@ const getPluralizationKey = (count: number): keyof PluralizedEntry =>
 const getEntry = (strings: Entries, key: string): Entry =>
   hasOwn(strings, key) ? strings[key] : key;
 
+/**
+ * Returns true if the given entry is a pluralization entry, or false otherwise.
+ *
+ * @param entry Entry to test.
+ *
+ * @return Whether entry is a pluralization entry.
+ */
+const isPluralizedEntry = (entry: Entry): entry is PluralizedEntry =>
+  typeof entry === 'object' && 'one' in (entry as PluralizedEntry);
```

Comment:
>do we need the casting on this second line?

Apparently not! Originally I did not include the first part of the condition (the `typeof` check), where it would have been necessary, but I guess that triggers TypeScript to narrow it enough for the `in` operator to be valid.